internal/idp/providers/saml/session.go

```diff
@@ -0,0 +1,90 @@
+package saml
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"net/url"
+
+	"github.com/crewjam/saml/samlsp"
+
+	"github.com/zitadel/zitadel/internal/errors"
+	"github.com/zitadel/zitadel/internal/idp"
+)
+
+var _ idp.Session = (*Session)(nil)
+
+// Session is the [idp.Session] implementation for the SAML provider.
+type Session struct {
+	ServiceProvider *samlsp.Middleware
+	state           string
+
+	RequestID string
+	Request   *http.Request
+}
+
+// GetAuthURL implements the [idp.Session] interface.
+func (s *Session) GetAuth(ctx context.Context) (string, bool) {
+	url, _ := url.Parse(s.state)
+	resp := NewTempResponseWriter()
+
+	request := &http.Request{
+		URL: url,
+	}
+	s.ServiceProvider.HandleStartAuthFlow(
+		resp,
+		request.WithContext(ctx),
+	)
+
+	if location := resp.Header().Get("Location"); location != "" {
+		return idp.Redirect(location)
+	}
+	return idp.Form(string(resp.content.Bytes()))
+}
```

Comment:
as already discussed: i still don't like it how this needs to be solved

and also unsure whether returning the whole form is a good idea

Reply:
Hm I still have not really a better idea, as the form is normally also returned as the response by the IDP, I don't think that it is a bad solution. At the same time we could also just return the information with RelayState, SAMLResponse and the URL in a map[string]string, but not really sure if that makes it better as then you have to build the request yourself.